plugins/node/opentelemetry-instrumentation-pg/src/utils.ts

```diff
@@ -289,3 +292,65 @@ export function patchClientConnectCallback(
     cb.call(this, err);
   };
 }
+
+function hasValidSqlComment(query: string): boolean {
+  const indexOpeningDashDashComment = query.indexOf('--');
```

Comment:
Does the same also apply for the block comment? i.e if it is inside of a string literal, it is not a comment? I took the current code almost verbatim from the existing sqlcommenter Node implementation (the [knex one for example](https://github.com/open-telemetry/opentelemetry-sqlcommenter/blob/2f8841add68358069ebf1c0ee560ab3e98a59aa9/nodejs/sqlcommenter-nodejs/packages/sqlcommenter-knex/util.js#L35-L55)), so the fault is present there as well.

Adding a parser would not really be a problem, I think, although not sure what the overhead would be for each query execution, perhaps a comment + a OpenTelemetry diagnostic log line for queries that are ignored?

Reply:
Yes, it might affect a block comment also. I agree the issue originates in the original SQLCommenter libraries. That's why i am inclined towards adding a comment regarding this issue here (and if possible callout in docs). 

Adding a parser might affect the performance as well as will be an additional dependency in this library (which databases are anyway doing). 

We are thinking if the specs regarding "do not add a comment if there is already a comment" could be modified. Should it just be "add a new comment at the end of the statement" ?. 
But, for that we need to understand whether different databases (at-least the supported ones Postgres, MySQL, SQLite, etc) have any constraints on comments and whether any popular "hint"(Ex [pg_hint_plan](https://pghintplan.osdn.jp/pg_hint_plan.html), [MySQL query hints](https://dev.mysql.com/doc/refman/5.7/en/optimizer-hints.html)) features will get affected if multiple comment sections are present.

So, maybe till that is finalised, we can add a comment and tech debt maybe ?

Reply:
Added a comment and a short note to the README for the flag.